src/main/java/com/commercetools/sync/categories/utils/CategorySyncUtils.java

```diff
@@ -0,0 +1,192 @@
+        final List<UpdateAction<Category>> coreActions = buildCoreActions(oldCategory, newCategory, syncOptions);
+        final List<UpdateAction<Category>> assetUpdateActions = buildAssetActions(oldCategory, newCategory, syncOptions);
+        final List<UpdateAction<Category>> allUpdateActions = Stream.concat(coreActions.stream(),
+                assetUpdateActions.stream())
+                .collect(Collectors.toList());
+        return filterUpdateActions(allUpdateActions, syncOptions.getUpdateActionsFilter());
+    }
+
+    /**
+     * Compares the Name, Slug, Description, Parent, OrderHint, MetaTitle, MetaDescription, MetaKeywords and Custom
+     * fields/ type fields of a {@link Category} and a {@link CategoryDraft}. It returns a {@link List} of
+     * {@link UpdateAction<Category>} as a result. If no update action is needed, for example in
+     * case where both the {@link Category} and the {@link CategoryDraft} have the same parents, an empty
+     * {@link List} is returned.
+     *
+     * @param oldCategory the category which should be updated.
+     * @param newCategory the category draft where we get the new data.
+     * @param syncOptions the sync options wrapper which contains options related to the sync process supplied by the user.
+     *                    For example, custom callbacks to call in case of warnings or errors occurring on the build
+     *                    update action process. And other options (See {@link BaseSyncOptions}
+     *                    for more info.
+     * @return A list of category-specific update actions.
+     */
+    @Nonnull
+    public static List<UpdateAction<Category>> buildCoreActions(@Nonnull final Category oldCategory,
+                                                                @Nonnull final CategoryDraft newCategory,
+                                                                @Nonnull final CategorySyncOptions syncOptions) {
+        final List<UpdateAction<Category>> updateActions = new ArrayList<>();
+        buildChangeNameUpdateAction(oldCategory, newCategory)
+                .map(updateActions::add);
+
+        buildChangeSlugUpdateAction(oldCategory, newCategory)
+                .map(updateActions::add);
+
+        buildSetDescriptionUpdateAction(oldCategory, newCategory, syncOptions)
+                .map(updateActions::add);
+
+        buildChangeParentUpdateAction(oldCategory, newCategory, syncOptions)
+                .map(updateActions::add);
+
+        buildChangeOrderHintUpdateAction(oldCategory, newCategory, syncOptions)
+                .map(updateActions::add);
+
+        buildSetMetaTitleUpdateAction(oldCategory, newCategory)
+                .map(updateActions::add);
+
+        buildSetMetaDescriptionUpdateAction(oldCategory, newCategory)
+                .map(updateActions::add);
+
+        buildSetMetaKeywordsUpdateAction(oldCategory, newCategory)
+                .map(updateActions::add);
```

Comment:
The repetitions are one liners. Having it as a function, would mean calling it 8 times as well. Maybe I can't imagine what you mean here. Also not sure how this could be a loop.